[Kenneth Seet] iP

itstrueitstrueitsrealitsreal opened the pull request:
# Kenneth's iP Pull Request
I made some changes to the original _Duke_ template, namely:
- changing _Duke_ to _Naruto_
- making it _Naruto_ themed :shipit: 🍥 

>“I'll become a Hokage someday”
> ~[Naruto](https://www.google.com/url?sa=t&rct=j&q=&esrc=s&source=web&cd=&cad=rja&uact=8&ved=2ahUKEwib5dmT6oyEAxU_9zgGHaGXBzkQFnoECBIQAw&url=https%3A%2F%2Fwww.sportskeeda.com%2Fanime%2F10-best-naruto-uzumaki-quotes-one-can-forget%23%3A~%3Atext%3D%25E2%2580%259CI%27ll%2520become%2520a%2520Hokage%2Care%2520popular%2520in%2520the%2520series&usg=AOvVaw15zptVC9VMQtWAj4wPmBaR&opi=89978449)

I also added the following classes of objects:
1. Tasks
2. Actions
3. Exceptions

The current structure of the project is as follows:
```
.
├── bin
│   ├── action
│   ├── exception
│   ├── main
│   │   ├── action
│   │   ├── exception
│   │   ├── naruto
│   │   ├── task
│   │   └── util
│   ├── naruto
│   ├── task
│   ├── test
│   │   ├── action
│   │   ├── exception
│   │   ├── naruto
│   │   ├── task
│   │   └── util
│   └── util
├── build
│   ├── classes
│   │   └── java
│   │       └── main
│   │           ├── action
│   │           ├── exception
│   │           ├── naruto
│   │           ├── task
│   │           └── util
│   ├── generated
│   │   └── sources
│   │       ├── annotationProcessor
│   │       │   └── java
│   │       │       └── main
│   │       └── headers
│   │           └── java
│   │               └── main
│   ├── libs
│   │   └── src
│   │       └── logs
│   └── tmp
│       ├── compileJava
│       └── shadowJar
├── config
│   └── checkstyle
├── docs
├── gradle
│   └── wrapper
├── out
│   └── production
│       └── ip
│           ├── action
│           ├── exception
│           ├── naruto
│           ├── task
│           └── util
├── src
│   ├── logs
│   ├── main
│   │   ├── bin
│   │   └── java
│   │       ├── action
│   │       ├── exception
│   │       ├── naruto
│   │       ├── task
│   │       └── util
│   └── test
│       └── java
│           ├── action
│           ├── exception
│           ├── naruto
│           ├── task
│           └── util
└── text-ui-test
    └── src
        └── logs
```
These are the things that still have to be done:
- [x] Update the description of my pull request
- [ ] Make a GUI for the iP using JavaFX
- [ ] Make use of VarArgs for the search function


Look at how compact my Main class is!
```java
import naruto.Naruto;
/**
 * The Main class represents the entry point of the Naruto chatbot.
 * It contains the main method that starts the execution of the program.
 */
public class Main {
    public static void main(String[] args) {
        new Naruto();
        while (Naruto.hasNextAction()) {
            Naruto.act();
            Naruto.listen();
        }
    }
}
```

> [!NOTE]
> I didn't watch Naruto.

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

itstrueitstrueitsrealitsreal committed:
completed Level 0

itstrueitstrueitsrealitsreal committed:
completed Level-1

itstrueitstrueitsrealitsreal committed:
reorganised project into Main, Naruto and Printutil

itstrueitstrueitsrealitsreal committed:
added Action interface, created Goodbye, Greet, Echo classes

itstrueitstrueitsrealitsreal committed:
changed echo to add, added store and list

itstrueitstrueitsrealitsreal committed:
added task class, mark and unmark actions

itstrueitstrueitsrealitsreal committed:
touched up aesthetics

itstrueitstrueitsrealitsreal committed:
added deadline, event, todo as types of tasks

itstrueitstrueitsrealitsreal committed:
updated text-ui-text

itstrueitstrueitsrealitsreal committed:
changed array to arraylist in store

itstrueitstrueitsrealitsreal committed:
organised files into packages, updated runtest files, added exceptions

itstrueitstrueitsrealitsreal committed:
added invalidindexexception and updated test input

itstrueitstrueitsrealitsreal committed:
Level-6: added delete functionality

itstrueitstrueitsrealitsreal committed:
implemented delete functionality, collections and enums

itstrueitstrueitsrealitsreal committed:
fixed checkstyle

itstrueitstrueitsrealitsreal committed:
removed unused imports

itstrueitstrueitsrealitsreal committed:
Add store functionality

itstrueitstrueitsrealitsreal committed:
Merge branch 'branch-Level-7'

itstrueitstrueitsrealitsreal committed:
Update .gitignore

itstrueitstrueitsrealitsreal committed:
Remove old logs file

itstrueitstrueitsrealitsreal committed:
Update .gitignore

itstrueitstrueitsrealitsreal committed:
Add date functionality

itstrueitstrueitsrealitsreal committed:
Fix bug in Store.java

itstrueitstrueitsrealitsreal committed:
Merge branch 'branch-Level-8'

itstrueitstrueitsrealitsreal committed:
Separate Store into Storage and TaskList

itstrueitstrueitsrealitsreal committed:
Add Ui

itstrueitstrueitsrealitsreal committed:
Separate Naruto::listen() into Ui and Parser

itstrueitstrueitsrealitsreal committed:
Touch up Action.java

itstrueitstrueitsrealitsreal committed:
Merge branch 'branch-A-More-OOP'

itstrueitstrueitsrealitsreal committed:
Merge remote-tracking branch 'remotes/origin/add-gradle-support'

itstrueitstrueitsrealitsreal committed:
Add Gradle support and Checkstyle

itstrueitstrueitsrealitsreal committed:
Touch up checkstyle

itstrueitstrueitsrealitsreal committed:
Add JUnit tests for CsvUtil and Parser

itstrueitstrueitsrealitsreal committed:
Merge branch 'branch-A-JUnit'

itstrueitstrueitsrealitsreal committed:
Add JAR

itstrueitstrueitsrealitsreal committed:
Add find

itstrueitstrueitsrealitsreal committed:
Touch up checkstyle

itstrueitstrueitsrealitsreal committed:
Add JavaDoc

itstrueitstrueitsrealitsreal committed:
Update gradle.build

itstrueitstrueitsrealitsreal committed:
Merge branch 'branch-A-CodingStandard'

itstrueitstrueitsrealitsreal committed:
Merge branch 'branch-A-JavaDoc'

itstrueitstrueitsrealitsreal committed:
Merge branch-Level-9 with master

itstrueitstrueitsrealitsreal committed:
Merge branch-Level-9 with master

itstrueitstrueitsrealitsreal committed:
Merge branch 'master' of https://github.com/itstrueitstrueitsrealitsreal/ip

itstrueitstrueitsrealitsreal committed:
Fix JAR generation

itstrueitstrueitsrealitsreal committed:
Add search for keywords out of order using varargs

itstrueitstrueitsrealitsreal committed:
Merge branch 'branch-A-Varargs'

itstrueitstrueitsrealitsreal committed:
Add GUI

itstrueitstrueitsrealitsreal committed:
Complete styling

itstrueitstrueitsrealitsreal committed:
Complete GUI

itstrueitstrueitsrealitsreal committed:
Merge branch 'branch-Level-10'

itstrueitstrueitsrealitsreal committed:
Fix bye command and JavaDoc

itstrueitstrueitsrealitsreal committed:
Fix bye message

itstrueitstrueitsrealitsreal committed:
Fix background color and dialogBox stretch issue

KhoonSun47 commented on src/main/java/action/Unmark.java:
```diff
@@ -0,0 +1,38 @@
+    @Override
+    public String execute() throws IOException {
+        StringBuilder message = new StringBuilder();
+        message.append("You know, sometimes things don't go as planned, but that's okay! ")
```
The line length of all your code is good, not too long.

KhoonSun47 commented on src/main/java/util/DateTimeUtil.java:
```diff
@@ -0,0 +1,63 @@
+ * The DateTimeUtil class provides utility methods for formatting and parsing date and time.
+ */
+public class DateTimeUtil {
+    private static final DateTimeFormatter input = DateTimeFormatter.ofPattern("d/MM/yyyy HHmm");
```
I think constant names should be all uppercase using underscore to separate words, following the Java coding standard.

KhoonSun47 commented on src/main/java/util/Ui.java:
```diff
@@ -0,0 +1,101 @@
+        String description;
+        int idx;
+        assert restOfLine != null;
+        switch (input) {
```
I like that you have configured your IDE to ensure that the switch statement does not have any indentation for the case clause, nice!

KhoonSun47 commented on src/main/java/util/Ui.java:
```diff
@@ -0,0 +1,101 @@
+
+    }
+
+    /**
```
It is good that you ensure all your public classes and methods have Javadoc comments, and they follow the conventions of the Java coding standard, nice!

KhoonSun47 commented on src/main/java/util/Storage.java:
```diff
@@ -0,0 +1,90 @@
+package util;
+import java.io.FileWriter;
```
Should your import statements be consistent throughout the project, some have one-line spacings between the package and import, while some do not.

itstrueitstrueitsrealitsreal replied:
Good catch, didn't think of that!

KhoonSun47 reviewed (commented):
Great job overall! Just a few minor adjustments needed to align with Java Coding Standards. Keep up the excellent work!

garywongkai commented on src/main/java/action/Mark.java:
```diff
@@ -0,0 +1,38 @@
+ */
+public class Mark implements Action {
+    private final TaskList taskList;
+    private final int i;
```
Can consider changing the naming convention for this, even though its only used as indexing.

garywongkai commented on src/main/java/naruto/Naruto.java:
```diff
@@ -0,0 +1,73 @@
+ * The Naruto class represents the main class of the Naruto application.
+ * It manages the list of actions to be performed and handles exceptions.
+ */
+public class Naruto {
```
AMAZING class name 💯 💯 👍  HAHAHAHA

garywongkai commented on src/main/java/task/Deadline.java:
```diff
@@ -0,0 +1,75 @@
+ * Represents a task with a deadline.
+ */
+public class Deadline extends Task {
+    protected LocalDateTime by;
```
maybe consider changing "by" to "by_date" to improve clarity!

garywongkai commented on src/main/java/util/CsvUtil.java:
```diff
@@ -0,0 +1,137 @@
+            return new Event(isMarked, description, DateTimeUtil.parse(from),
+                    DateTimeUtil.format(to));
+        default:
+            throw new IllegalArgumentException("Invalid 'event' value: " + event);
```
great use of error handling when input is not what is expected :)

garywongkai commented on src/main/resources/images/user.png:
👍

garywongkai commented on src/main/java/util/Ui.java:
```diff
@@ -0,0 +1,101 @@
+            }
+            return new Unmark(taskList, idx);
+        case "todo":
+            try {
+                description = Parser.parseDescription(restOfLine);
+            } catch (NarutoException e) {
+                return new HandleError(e);
+            }
+            return new Add(new ToDo(description), taskList);
+        case "deadline":
+            try {
+                tokens = Parser.parseDeadline(restOfLine);
+            } catch (NarutoException e) {
+                return new HandleError(e);
+            }
+            return new Add(new Deadline(tokens[0], DateTimeUtil.format(tokens[1])), taskList);
+        case "event":
+            try {
+                tokens = Parser.parseEvent(restOfLine);
+            } catch (NarutoException e) {
+                return new HandleError(e);
+            }
+            return new Add(new Event(tokens[0], DateTimeUtil.format(tokens[1]),
+                DateTimeUtil.format(tokens[2])), taskList);
+        case "delete":
+            try {
+                idx = Parser.parseIdx(restOfLine, taskList);
+            } catch (NarutoException e) {
+                return new HandleError(e);
+            }
+            return new Delete(taskList, idx);
+        case "find":
+            try {
+                tokens = Parser.parseDescription(restOfLine).split(" ");
+            } catch (NarutoException e) {
+                return new HandleError(e);
+            }
+            return new Find(tokens);
+        default:
+            return new HandleError(NarutoException.createInvalidCommandException());
```
Great error handling with all these try-catch blocks!

garywongkai commented on src/main/java/exception/NarutoException.java:
```diff
@@ -0,0 +1,128 @@
+    /**
+     * Enum representing the different types of errors that can occur in the Naruto application.
+     */
+    public enum ErrorType {
+        EMPTY_COMMAND,
+        EMPTY_DEADLINE,
+        EMPTY_EVENT,
+        INVALID_DEADLINE,
+        INVALID_EVENT,
+        INVALID_COMMAND,
+        INVALID_INDEX,
+        FILE_CORRUPTED,
+        INVALID_ACTION
+    }
```
Great use of enum for errors!
Using an enum here to represent error types really improves the code readability :))

garywongkai reviewed (commented):
Great job on your program really :), nothing much to comment about other than naming convention for variables and improvements to code readability!

itstrueitstrueitsrealitsreal committed:
Fix checkstyle

itstrueitstrueitsrealitsreal committed:
Add assertions to Parser, Naruto, and Actions

Assertions have been added to the Parser, Naruto, and Action classes in order for ease
of bug detection during testing.

itstrueitstrueitsrealitsreal committed:
Fix access of static variable in TaskList
- Previously, `this.tasks` was used in TaskList.java
- Now, `TaskList.tasks` is being used to make sure that the static variable is being accessed

itstrueitstrueitsrealitsreal committed:
Fix access of static variable in TaskList
- Previously, `this.tasks` was used in TaskList.java
- Now, `TaskList.tasks` is being used to make sure that the static variable is being accessed

itstrueitstrueitsrealitsreal committed:
Improve code quality
- Standardised newline between package names and imports
- Updated variable names to be more informative
- Updated variable names of constants

itstrueitstrueitsrealitsreal committed:
Merge branch 'A-Streams' of https://github.com/itstrueitstrueitsrealitsreal/ip into A-Streams

itstrueitstrueitsrealitsreal committed:
Merge pull request #1 from itstrueitstrueitsrealitsreal/A-Assertions

Add assertions to Parser, Naruto, and Actions

itstrueitstrueitsrealitsreal committed:
Merge branch 'master' into A-Streams

itstrueitstrueitsrealitsreal committed:
Merge branch 'master' into A-CodeQuality

itstrueitstrueitsrealitsreal committed:
Merge pull request #3 from itstrueitstrueitsrealitsreal/A-Streams

A streams

itstrueitstrueitsrealitsreal committed:
Merge branch 'master' into A-CodeQuality

itstrueitstrueitsrealitsreal committed:
Merge pull request #2 from itstrueitstrueitsrealitsreal/A-CodeQuality

Improve code quality

itstrueitstrueitsrealitsreal committed:
Add CI
Added .yml file according to guide on CS2103 website.

itstrueitstrueitsrealitsreal committed:
Clean up code
- Removed unused variables
- Removed unused imports

itstrueitstrueitsrealitsreal committed:
Fix bug

itstrueitstrueitsrealitsreal committed:
Update variable name
- Changed i to idx

itstrueitstrueitsrealitsreal committed:
Merge pull request #6 from itstrueitstrueitsrealitsreal/branch-A-Assertions

Update variable name

itstrueitstrueitsrealitsreal committed:
Merge branch 'master' into branch-A-Streams

itstrueitstrueitsrealitsreal committed:
Merge pull request #5 from itstrueitstrueitsrealitsreal/branch-A-Streams

Fix bug

itstrueitstrueitsrealitsreal committed:
Merge branch 'master' into branch-A-CodeQuality

itstrueitstrueitsrealitsreal committed:
Merge pull request #4 from itstrueitstrueitsrealitsreal/branch-A-CodeQuality

Clean up code

itstrueitstrueitsrealitsreal committed:
Merge branch 'master' into branch-A-CodeQuality

itstrueitstrueitsrealitsreal committed:
Fix checkstyle
- Fixed checkstyle error that lead to CI tests failing

itstrueitstrueitsrealitsreal committed:
Update gradle.yml

itstrueitstrueitsrealitsreal committed:
Implement sort

itstrueitstrueitsrealitsreal committed:
Fix checkstyle

itstrueitstrueitsrealitsreal committed:
Merge pull request #7 from itstrueitstrueitsrealitsreal/C-Sort

Implement sort

itstrueitstrueitsrealitsreal committed:
Fix Code Quality

itstrueitstrueitsrealitsreal committed:
Update gradle.yml

itstrueitstrueitsrealitsreal committed:
Merge pull request #8 from itstrueitstrueitsrealitsreal/branch-A-CodeQuality

Branch a code quality

itstrueitstrueitsrealitsreal committed:
Add improved commands

itstrueitstrueitsrealitsreal committed:
Merge branch 'master' of https://github.com/itstrueitstrueitsrealitsreal/ip

itstrueitstrueitsrealitsreal committed:
Add help

itstrueitstrueitsrealitsreal committed:
Remove unprintable ascii

itstrueitstrueitsrealitsreal committed:
Merge pull request #9 from itstrueitstrueitsrealitsreal/branch-A-Personality

Remove unprintable ascii

itstrueitstrueitsrealitsreal committed:
Add padding for DialogBox

itstrueitstrueitsrealitsreal committed:
Merge pull request #10 from itstrueitstrueitsrealitsreal/branch-A-BetterGui

Add padding for DialogBox

itstrueitstrueitsrealitsreal committed:
Add duplicate handling

itstrueitstrueitsrealitsreal committed:
Fix javadoc and messages

itstrueitstrueitsrealitsreal committed:
Merge pull request #11 from itstrueitstrueitsrealitsreal/branch-A-MoreErrorHandling

Add duplicate handling

itstrueitstrueitsrealitsreal committed:
Add ui

itstrueitstrueitsrealitsreal committed:
Add user guide

itstrueitstrueitsrealitsreal committed:
Update naming of .jar

itstrueitstrueitsrealitsreal committed:
Update README.md

itstrueitstrueitsrealitsreal committed:
Update README.md

itstrueitstrueitsrealitsreal committed:
Update README.md

itstrueitstrueitsrealitsreal committed:
Update README.md

itstrueitstrueitsrealitsreal committed:
Fix Ui.java

itstrueitstrueitsrealitsreal committed:
Fix checkstyle

itstrueitstrueitsrealitsreal committed:
Update README.md

itstrueitstrueitsrealitsreal committed:
Update README.md

itstrueitstrueitsrealitsreal committed:
Fix index bug

itstrueitstrueitsrealitsreal committed:
Update README.md

itstrueitstrueitsrealitsreal committed:
Update README.md

itstrueitstrueitsrealitsreal committed:
Update README.md

itstrueitstrueitsrealitsreal committed:
Add _config.yml
